python/xorbits/_mars/dataframe/indexing/tests/test_indexing_execution.py

```diff
@@ -1481,14 +1483,15 @@ def test_where_execution(setup):
     pd.testing.assert_frame_equal(new_df.execute().fetch(), raw_df.mask(raw_df < 0))
 
 
-def test_set_axis_execution(setup):
+@support_cuda
```

Comment:
What's the differences with `require_cudf` and `require_cupy`.

Reply:
This annotation means the test case can run on GPU, but it can also run on CPU.

So it won't skip the test case when `cupy` or `cudf` is missing.

Reply:
How `setup_gpu` works on CPU?

Reply:
The difference between the two fixtures is that `setup_gpu` searches for cuda devices. But when there's no available cuda device, `setup_gpu` won't raise an Exception, instead it will pass an empty list as the parameter `cuda_devices`.

Reply:
So, `setup_gpu` is same as `setup` with no cuda device available?

Reply:
> So, `setup_gpu` is same as `setup` with no cuda device available?

As I understand it, when there are GPUs, `setup_gpu` discoveries available GPUs and init test runtime with them, and when there is no GPU, `setup_gpu` is equivalent to `setup`.